packages/tldraw/src/lib/ui/components/StylePanel/DefaultStylePanelContent.tsx

```diff
@@ -432,65 +414,28 @@ export function ArrowheadStylePickerSet({ styles }: StylePickerSetProps) {
 			styleB={ArrowShapeArrowheadEndStyle}
 			itemsB={STYLES.arrowheadEnd}
 			valueB={arrowheadEnd}
-			onValueChange={handleValueChange}
 			labelA="style-panel.arrowhead-start"
 			labelB="style-panel.arrowhead-end"
 		/>
 	)
 }
 
-const tldrawSupportedOpacities = [0.1, 0.25, 0.5, 0.75, 1] as const
 /** @public @react */
-export function OpacitySlider() {
-	const editor = useEditor()
-
-	const onHistoryMark = useCallback((id: string) => editor.markHistoryStoppingPoint(id), [editor])
-	const showUiLabels = useValue('showUiLabels', () => editor.user.getShowUiLabels(), [editor])
-
-	const opacity = useValue('opacity', () => editor.getSharedOpacity(), [editor])
-	const trackEvent = useUiEvents()
-	const msg = useTranslation()
-
-	const handleOpacityValueChange = React.useCallback(
-		(value: number) => {
-			const item = tldrawSupportedOpacities[value]
-			editor.run(() => {
-				if (editor.isIn('select')) {
-					editor.setOpacityForSelectedShapes(item)
-				}
-				editor.setOpacityForNextShapes(item)
-				editor.updateInstanceState({ isChangingStyle: true })
-			})
-
-			trackEvent('set-style', { source: 'style-panel', id: 'opacity', value })
-		},
-		[editor, trackEvent]
-	)
-
-	if (opacity === undefined) return null
-
-	const opacityIndex =
-		opacity.type === 'mixed'
-			? -1
-			: tldrawSupportedOpacities.indexOf(
-					minBy(tldrawSupportedOpacities, (supportedOpacity) =>
-						Math.abs(supportedOpacity - opacity.value)
-					)!
-				)
+export function StylePanelSplinePicker() {
+	const { styles } = useStylePanelContext()
+	const spline = styles.get(LineShapeSplineStyle)
+	if (spline === undefined) return null
 
 	return (
-		<>
-			{showUiLabels && <StylePanelSubheading>{msg('style-panel.opacity')}</StylePanelSubheading>}
```

Comment:
looks like we lost the opacity ui label, oops. i hope that doesn't mess up your complicated CSS selector 😆